lib/chiapos/src/aes.hpp

```diff
@@ -1,270 +1,484 @@
-        m = _mm_aesenc_si128(m, k[4]);      \
-        m = _mm_aesenc_si128(m, k[5]);      \
-        m = _mm_aesenc_si128(m, k[6]);      \
-        m = _mm_aesenc_si128(m, k[7]);      \
-        m = _mm_aesenc_si128(m, k[8]);      \
-        m = _mm_aesenc_si128(m, k[9]);      \
-        m = _mm_aesenclast_si128(m, k[10]); \
-    } while (0)
-
-/**
- *  Encrypts a message of 128 bits with a 256 bit key, using
- *  13 rounds of AES256 (13 full rounds and one final round). Uses
- *  AES-NI assembly instructions.
- */
-#define DO_ENC_BLOCK_256(m, k) \
-    do {\
-        m = _mm_xor_si128(m, k[ 0]); \
-        m = _mm_aesenc_si128(m, k[ 1]); \
-        m = _mm_aesenc_si128(m, k[ 2]); \
-        m = _mm_aesenc_si128(m, k[ 3]); \
-        m = _mm_aesenc_si128(m, k[ 4]); \
-        m = _mm_aesenc_si128(m, k[ 5]); \
-        m = _mm_aesenc_si128(m, k[ 6]); \
-        m = _mm_aesenc_si128(m, k[ 7]); \
-        m = _mm_aesenc_si128(m, k[ 8]); \
-        m = _mm_aesenc_si128(m, k[ 9]); \
-        m = _mm_aesenc_si128(m, k[ 10]);\
-        m = _mm_aesenc_si128(m, k[ 11]);\
-        m = _mm_aesenc_si128(m, k[ 12]);\
-        m = _mm_aesenc_si128(m, k[ 13]);\
-        m = _mm_aesenclast_si128(m, k[ 14]);\
-    }while(0)
-
-/**
- * Encrypts a message of 128 bits with a 128 bit key, using
- * 2 full rounds of AES128. Uses AES-NI assembly instructions.
- */
-#define DO_ENC_BLOCK_2ROUND(m, k)      \
-    do                                 \
-    {                                  \
-        m = _mm_xor_si128(m, k[0]);    \
-        m = _mm_aesenc_si128(m, k[1]); \
-        m = _mm_aesenc_si128(m, k[2]); \
-    } while (0)
-/**
- * Decrypts a ciphertext of 128 bits with a 128 bit key, using
- * 10 rounds of AES128 (9 full rounds and one final round).
- * Uses AES-NI assembly instructions.
- */
-#define DO_DEC_BLOCK(m, k)                  \
-    do                                      \
-    {                                       \
-        m = _mm_xor_si128(m, k[10 + 0]);    \
-        m = _mm_aesdec_si128(m, k[10 + 1]); \
-        m = _mm_aesdec_si128(m, k[10 + 2]); \
-        m = _mm_aesdec_si128(m, k[10 + 3]); \
-        m = _mm_aesdec_si128(m, k[10 + 4]); \
-        m = _mm_aesdec_si128(m, k[10 + 5]); \
-        m = _mm_aesdec_si128(m, k[10 + 6]); \
-        m = _mm_aesdec_si128(m, k[10 + 7]); \
-        m = _mm_aesdec_si128(m, k[10 + 8]); \
-        m = _mm_aesdec_si128(m, k[10 + 9]); \
-        m = _mm_aesdeclast_si128(m, k[0]);  \
-    } while (0)
-
-/**
- * Decrypts a ciphertext of 128 bits with a 128 bit key, using
- * 2 full rounds of AES128. Uses AES-NI assembly instructions.
- * Will not work unless key schedule is modified.
- */
 /*
-#define DO_DEC_BLOCK_2ROUND(m, k)           \
-    do                                      \
-    {                                       \
-        m = _mm_xor_si128(m, k[2 + 0]);    \
-        m = _mm_aesdec_si128(m, k[2 + 1]); \
-        m = _mm_aesdec_si128(m, k[2 + 2]); \
-    } while (0)
+
```

Comment:
We need to add the open source license for this file.

Reply:
Added public domain comment. Thanks!